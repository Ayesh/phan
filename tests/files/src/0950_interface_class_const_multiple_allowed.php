<?php

namespace NS950;

interface I1 {
    const A = 1;
}

interface I2 extends I1 {
}

class X implements I1, I2 { // This is allowed
}

interface I3 extends I2 {
    const A = 1;  // This is forbidden
}
var_dump(new X());
