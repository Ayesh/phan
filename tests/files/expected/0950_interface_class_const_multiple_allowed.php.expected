%s:16 PhanAccessOverridesFinalConstant Declaration of class constant \NS950\I3::A overrides final constant \NS950\I1::A defined at %s:6
